client/oci_images.go

```diff
@@ -0,0 +1,312 @@
+	"github.com/lxc/incus/v6/shared/osarch"
+	"github.com/lxc/incus/v6/shared/subprocess"
+)
+
+type ociInfo struct {
+	Alias        string
+	Name         string    `json:"Name"`
+	Digest       string    `json:"Digest"`
+	Created      time.Time `json:"Created"`
+	Architecture string    `json:"Architecture"`
+	LayersData   []struct {
+		Size int64 `json:"Size"`
+	} `json:"LayersData"`
+}
+
+// Image handling functions
+
+// GetImages returns a list of available images as Image structs.
+func (r *ProtocolOCI) GetImages() ([]api.Image, error) {
+	return nil, fmt.Errorf("Can't list images from OCI registry")
+}
+
+// GetImagesAllProjects returns a list of available images as Image structs.
+func (r *ProtocolOCI) GetImagesAllProjects() ([]api.Image, error) {
+	return nil, fmt.Errorf("Can't list images from OCI registry")
+}
+
+// GetImageFingerprints returns a list of available image fingerprints.
+func (r *ProtocolOCI) GetImageFingerprints() ([]string, error) {
+	return nil, fmt.Errorf("Can't list images from OCI registry")
+}
+
+// GetImagesWithFilter returns a filtered list of available images as Image structs.
+func (r *ProtocolOCI) GetImagesWithFilter(filters []string) ([]api.Image, error) {
+	return nil, fmt.Errorf("Can't list images from OCI registry")
+}
+
+// GetImage returns an Image struct for the provided fingerprint.
+func (r *ProtocolOCI) GetImage(fingerprint string) (*api.Image, string, error) {
+	info, ok := r.cache[fingerprint]
+	if !ok {
+		return nil, "", fmt.Errorf("Image not found")
+	}
+
+	img := api.Image{
+		ImagePut: api.ImagePut{
+			Public: true,
+		},
+		Aliases: []api.ImageAlias{{
+			Name: info.Alias,
+		}},
+		Architecture: info.Architecture,
+		Fingerprint:  fingerprint,
+		Type:         string(api.InstanceTypeContainer),
+		CreatedAt:    info.Created,
+		UploadedAt:   info.Created,
+	}
+
+	var size int64
+	for _, layer := range info.LayersData {
+		size += layer.Size
+	}
+
+	img.Size = size
+
+	return &img, "", nil
+}
+
+// GetImageFile downloads an image from the server, returning an ImageFileResponse struct.
+func (r *ProtocolOCI) GetImageFile(fingerprint string, req ImageFileRequest) (*ImageFileResponse, error) {
+	info, ok := r.cache[fingerprint]
+	if !ok {
+		return nil, fmt.Errorf("Image not found")
+	}
+
+	// Quick checks.
+	if req.MetaFile == nil && req.RootfsFile == nil {
+		return nil, fmt.Errorf("No file requested")
+	}
+
+	if os.Geteuid() != 0 {
+		return nil, fmt.Errorf("OCI image export currently requires root access")
+	}
+
+	// Get some temporary storage.
+	ociPath, err := os.MkdirTemp("", "incus-oci-")
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() { _ = os.RemoveAll(ociPath) }()
+
+	err = os.Mkdir(filepath.Join(ociPath, "oci"), 0700)
+	if err != nil {
+		return nil, err
+	}
+
+	err = os.Mkdir(filepath.Join(ociPath, "image"), 0700)
+	if err != nil {
+		return nil, err
+	}
+
+	// Copy the image.
+	_, err = subprocess.RunCommand(
+		"skopeo",
+		"--insecure-policy",
+		"copy",
```

Comment:
Is there any reason `skopeo` and `umoci` is called as sub-process instead of used as a library?

Reply:
Mostly simplicity at this stage. Neither appear to be particularly designed to be consumed as Go modules and we also don't want to find ourselves importing too many indirect Go dependencies into the core of our project.

As both `skopeo` and `umoci` are readily packaged in most distributions, just using the external commands is best for now.

Reply:
Would it be possible to get new commands/dependencies added to the release notes? Or maybe keep a running list in the packaging doc? 

(This is amazing btw :)